### PR TITLE
Sage: Integrate Pine into icons

### PR DESCRIPTION
**Pull request by monicawheeler**
## Description
Updates components and documentation to use `pds-icon`. Switching over the new icon system should help to eliminate the current manual process.

## Screenshots
Minimal visual changes should be expected.

## Testing in `sage-lib`

- Navigate to each component for [Rails](http://localhost:4000/pages/index) & [React](http://localhost:4100/?path=/docs/welcome-to-sage--page)
- Verify icons render as expected.
- Verify icons render in docs as expected

## Testing in `kajabi-products`

1. (**BREAKING**) Converts icon font to pds-icons in components and documentation.
   - [ ] A full regression of the kajabi admin should be performed to make sure all icons still display as expected.

## Related
[DSS-623](https://kajabi.atlassian.net/browse/DSS-623)

[DSS-623]: https://kajabi.atlassian.net/browse/DSS-623?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/app/views/application/_app_head_content.html.erb [hunk 1]
```diff
@@ -16,3 +16,5 @@
 <link href="/favicon-16x16.png" rel="icon" sizes="16x16" type="image/png">
 <link href="/favicon.ico" rel="icon" type="image/x-icon">
 <%= csrf_meta_tags %>
+<script type="module" src="https://cdn.jsdelivr.net/npm/@pine-ds/core/dist/pine-core/pine-core.esm.js"></script>
```
Comment (monicawheeler) [inline docs/app/views/application/_app_head_content.html.erb:19]: Adds the pine icon library in rails in order to serve the web components.

File: package.json [hunk 2]
```diff
@@ -71,5 +71,8 @@
     "watch:assets": "cd packages/sage-assets && yarn build:watch",
     "watch:react": "cd packages/sage-react && yarn build:watch",
     "watch:system": "cd packages/sage-system && yarn build:watch"
+  },
+  "dependencies": {
+    "@pine-ds/icons": "^4.2.0"
```
Comment (monicawheeler) [inline package.json:76]: Adds pine-ds/icons as a sage-react dependency

File: packages/sage-react/.storybook/manager-head.html [hunk 3]
```diff
@@ -9,3 +9,6 @@
   type="image/png"
   href="favicon.ico"
 />
+
+<script type="module" src="http://localhost:7200/build/pds-icons.esm.js"></script>
```
Comment (monicawheeler) [inline packages/sage-react/.storybook/manager-head.html:13]: brings in the locally run pds-icons files

File: packages/sage-react/lib/Icon/Icon.jsx [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import classnames from 'classnames';
+import { downSmall } from '@pine-ds/icons/icons';
```
Comment (monicawheeler) [inline packages/sage-react/lib/Icon/Icon.jsx:4]: tests a specific named icon - likely not necessary but used during testing

File: packages/sage-react/lib/Icon/Icon.jsx [hunk 5]
```diff
@@ -46,8 +47,18 @@ export const Icon = ({
     attributes.role = 'img';
   }
 
+  // const sizeMapping = {
```
Comment (monicawheeler) [inline packages/sage-react/lib/Icon/Icon.jsx:50]: commented out but left to ensure we account for differences between what pine is using and what sage react needs - this needs to be addressed

File: packages/sage-react/lib/Icon/Icon.jsx [hunk 6]
```diff
@@ -46,8 +47,18 @@ export const Icon = ({
...
+  //   [ICON_SIZES.XS]: 'small',
+  //   [ICON_SIZES.SM]: 'normal',
+  //   [ICON_SIZES.MD]: 'medium',
+  //   [ICON_SIZES.LG]: 'large',
+  //   [ICON_SIZES.XL]: 'large',
+  // };
+
   const renderIcon = () => (
-    <i className={classNames} {...attributes} {...rest} />
+    <i>
```
Comment (monicawheeler) [inline packages/sage-react/lib/Icon/Icon.jsx:59]: wrapped in `i` to test - not necessarily needed

File: packages/sage-react/lib/Icon/Icon.jsx [hunk 7]
```diff
@@ -46,8 +47,18 @@ export const Icon = ({
...
+      <pds-icon name={downSmall} />
```
Comment (monicawheeler) [inline packages/sage-react/lib/Icon/Icon.jsx:60]: testing a specific icon name from the import - this was in effort to troubleshoot the pds-icon not outputting the shadow root

File: packages/sage-react/package.json [hunk 8]
```diff
@@ -83,7 +83,8 @@
     "webpack-cli": "^3.3.12"
   },
   "dependencies": {
-    "@kajabi/sage-assets": "^1.14.4",
+    "@kajabi/sage-assets": "^1.14.5",
```
Comment (monicawheeler) [inline packages/sage-react/package.json:86]: updating for sage bump

File: packages/sage-react/package.json [hunk 9]
```diff
@@ -83,7 +83,8 @@
...
+    "@pine-ds/icons": "^4.2.0",
```
Comment (monicawheeler) [inline packages/sage-react/package.json:87]: added to utilize pine-ds/icons 🤔 - is this necessary if we are importing the js files directly into the manager-head.thml

Comment (monicawheeler) [reply packages/sage-react/package.json:87]: I am looking for info, not offering direction with this comment.

File: docs/app/views/application/_nav.html.erb [hunk 10]
```diff
@@ -48,7 +48,7 @@
         name: "stack"
       },
     } %>
-    <!-- COMING SOON 
+    <!-- COMING SOON
```
Comment (ju-Skinner) [inline docs/app/views/application/_nav.html.erb:51]: we need to close the comment ` -->`

Comment (pixelflips) [reply docs/app/views/application/_nav.html.erb:51]: i think it is closed, it's just a few lines down ~63.

Comment (ju-Skinner) [reply docs/app/views/application/_nav.html.erb:51]: Yep...you're right. I'm marking this resolved.

**Review by anechol — CHANGES_REQUESTED** [review]
Icons are coming in fine in Rails, but I can't see them on the React side. Additionally, because of them being web components, there is now a fraction of a second on page load with no cache where the icon isn't showing up. I can't imagine there will be complaints about it, but just making a note of it here, since this is being updated app-wide.

**Comment by pixelflips** [comment]
> Icons are coming in fine in Rails, but I can't see them on the React side. Additionally, because of them being web components, there is now a fraction of a second on page load with no cache where the icon isn't showing up. I can't imagine there will be complaints about it, but just making a note of it here, since this is being updated app-wide.

You will need to be running your local pine for the react side. Also, make sure the ports match your local instance in the `packages/sage-react/.storybook/preview-head.html` file.

**Review by QuintonJason — APPROVED** [review]
After changing line 86 of `packages/sage-react/package.json` to ` "@kajabi/sage-assets": "*",` and restarted the server, everything was good to go

**Comment by monicawheeler** [comment]
@pixelflips - the Category item page has an alert, http://localhost:4000/pages/component/catalog_item?tab=preview, is missing the icon 
| Local | Live |
|--------|--------|
| Screenshot 2024-05-30 at 2 15 41 PM | Screenshot 2024-05-30 at 2 15 45 PM |

**Comment by monicawheeler** [comment]
Property is missing the icons
http://localhost:4000/pages/component/property?tab=preview
Screenshot 2024-05-30 at 2 21 20 PM

**Comment by monicawheeler** [comment]
Typeahead is missing the icons in the dropdown
Screenshot 2024-05-30 at 2 22 54 PM

**Comment by pixelflips** [comment]
@monicawheeler those should all be addressed now, docs just had incorrect icon naming.

alert icons - done
property - done / forgot to push the commit for this one
typeahead - done

**Comment by monicawheeler** [comment]
Screenshot 2024-05-30 at 2 38 27 PM
On local, the arrows are not aligned properly in react

**Comment by pixelflips** [comment]
> On local, the arrows are not aligned properly in react

@monicawheeler before checking storybook, please apply the changes from quinton's comment above.

**Comment by monicawheeler** [comment]
Screenshot 2024-05-30 at 2 59 40 PM
Could we verify the button spacing between icons is consistent with the documentation? It appears wider in the local dev environment on react compared to the public url

**Comment by monicawheeler** [comment]
> > On local, the arrows are not aligned properly in react
> 
> @monicawheeler before checking storybook, please apply the `package.json` changes from quinton's comment above.

This is still an issue for me after pulling and updating the `packages/sage-react/package.json` file per Quinton's comments.

**Comment by monicawheeler** [comment]
> Screenshot 2024-05-30 at 2 59 40 PM Could we verify the button spacing between icons is consistent with the documentation? It appears wider in the local dev environment on react compared to the public url

This is resolved after pulling latest and following Quinton's `packages/sage-react/package.json` update.

File: docs/app/views/examples/components/panel/_preview.html.erb [hunk 11]
```diff
@@ -70,25 +70,25 @@ There are a few options for the panel body container.
   <%= sage_component SagePanelStack, {} do %>
     <%= sage_component SageCard, {} do %>
       <%= sage_component SageCardRow, { grid_template: "et" } do %>
-        <i class="sage-icon-video-on"></i>
+        <%= sage_component SageIcon, { icon: "video-on" } %>
```
Comment (ju-Skinner) [inline docs/app/views/examples/components/panel/_preview.html.erb:73]: Question why did we elect to use `SageIcon` here versus `pds-icon`? I see in some instances we went from `<i...` to `<pds-icon...`

Not blocking, but just wanted to call this out for consistency

File: package.json (hunk 2 above)
Comment (ju-Skinner) [inline package.json:76]: we may need to up this to the latest version of [Pine Icons](https://www.npmjs.com/package/@pine-ds/icons)...currently v6.0.0

File: packages/sage-react/lib/Dropdown/Dropdown.jsx [hunk 12]
```diff
@@ -80,6 +80,10 @@ export const Dropdown = ({
     let directionX = null;
     let directionY = null;
     const el = wrapperRef.current;
+
+    // if el is null, return
+    if (!el) return;
```
Comment (ju-Skinner) [inline packages/sage-react/lib/Dropdown/Dropdown.jsx:85]: Looks like we need to rebase...This has been handled elsewhere.

**Review by ju-Skinner — APPROVED** [review]

**Review by monicawheeler — COMMENTED** [review]
Since I authored this PR, I cannot Approve. I did not do the work here though and after a review, I add my ✅ - This is good to go and great work!

**Review by anechol — CHANGES_REQUESTED** [review]
Nice work getting these updated!

Can we update the Banner component as well? That icon is missing.

**Comment by pixelflips** [comment]
> Nice work getting these updated!
> 
> Can we update the Banner component as well? That icon is missing.

@anechol great catch! those should be updated now as well.

**Pull request merged**